src/SkeletalTrapezoidation.cpp

```diff
@@ -0,0 +1,3083 @@
+    Point to = source_segment.to();
+
+    // find starting edge
+    // find end edge
+    bool first = true;
+    bool seen_possible_start = false;
+    bool after_start = false;
+    bool ending_edge_is_set_before_start = false;
+    for (vd_t::edge_type* edge = cell.incident_edge(); edge != cell.incident_edge() || first; edge = edge->next())
+    {
+        if (edge->is_infinite())
+        {
+            first = false;
+            continue;
+        }
+        bool check_secondary_edge = true;
+        Point v0 = VoronoiUtils::p(edge->vertex0());
+        Point v1 = VoronoiUtils::p(edge->vertex1());
+        assert(!(v0 == to && v1 == from));
+        if (v0 == to
+            && !after_start // use the last edge which starts in source_segment.to
+        )
+        {
+            starting_vd_edge = edge;
+            seen_possible_start = true;
+            check_secondary_edge = false;
+        }
+        else if (seen_possible_start)
+            after_start = true;
+        if (v1 == from
+            && (!ending_vd_edge || ending_edge_is_set_before_start)
+        )
+        {
+            ending_edge_is_set_before_start = !after_start;
+            ending_vd_edge = edge;
+            check_secondary_edge = false;
+        }
+        if (false && check_secondary_edge && edge->is_secondary()
+               != LinearAlg2D::pointLiesOnTheRightOfLine(v1, from, to)
+            &&    LinearAlg2D::pointLiesOnTheRightOfLine(v0, from, to)
+            && (LinearAlg2D::getDist2FromLineSegment(v0, from, v1) <= 5 // TODO: magic value
+                || LinearAlg2D::getDist2FromLineSegment(v0, to, v1) <= 5)
+        )
+        { // edge crosses source segment
+            // TODO: handle the case where two consecutive line segments are collinear!
+            // that's the only case where a voronoi segment doesn't end in a polygon vertex, but goes though it
+            if (LinearAlg2D::pointLiesOnTheRightOfLine(VoronoiUtils::p(edge->vertex1()), source_segment.from(), source_segment.to()))
+            {
+                ending_vd_edge = edge;
+            }
+            else
+            {
+                starting_vd_edge = edge;
+            }
+            first = false;
+            continue;
+        }
+        first = false;
+    }
+    
+    assert(starting_vd_edge && ending_vd_edge);
+    assert(starting_vd_edge != ending_vd_edge);
+    
+    start_source_point = source_segment.to();
+    end_source_point = source_segment.from();
+}
+
+SkeletalTrapezoidation::SkeletalTrapezoidation(const Polygons& polys, float transitioning_angle
+, coord_t discretization_step_size
+, coord_t transition_filter_dist
+, coord_t beading_propagation_transition_dist
+)
+: polys(polys)
+, transitioning_angle(transitioning_angle)
+, discretization_step_size(discretization_step_size)
+, transition_filter_dist(transition_filter_dist)
+, beading_propagation_transition_dist(beading_propagation_transition_dist)
+{
+    init();
```

Comment:
... Yeah. Ohkay. Please put the contents of the `init` function in the constructor instead. That's what the constructor is for. Or maybe better: Allow filling the trapezoidation from a polygon after constructing it.